07_hello_friend/friend.rb

```diff
@@ -0,0 +1,12 @@
+class Friend
+  def greeting(who = nil)
+    string = "Hello!"
+    if who != nil
+      string[string.length - 1] = ","
+      string = string.concat(" ")
+      string = string.concat(who)
+      string = string.concat("!")
+    end
+    string
```

Comment:
You can re-write this in a single line using string interpolation and a ternary operator.